### DISCUSSION TITLE
The multi line Triple-slash Comment in _dmmf documention are split by `\n` text

### DISCUSSION THREAD

**Discussion by j-deng** [prisma/prisma#10661] [Q&A] [answered]
"@prisma/client": "^3.6.0"

If the comment in schema is:
```
/// test\ncomment
/// test too
```
the  documention in _dmmf is print as
```
test\
comment\ntest too
```

I want to ask is it excepted or not?

**Comment by austincrim** [comment]
Hey @j-deng 👋 !

This definitely seems unexpected. I'll take a look at filing an issue. Would you mind telling me what you are using the `_dmmf` comments for?

**Reply by j-deng** [reply]
@austincrim I just to parse the document annotations at startup, like what [typegraphql prisma](https://prisma.typegraphql.com/docs/advanced/hiding-field) do but not compile to ts client

**Reply by austincrim** [reply]
After discussing this with some of our engineers, it seems that this is most likely an issue in how the `dmmf` is being printed or displayed, not an issue with how we are generating it internally.

I won't be creating an issue for this, but if you have any further questions, you can let me know.

**Reply by j-deng** [reply]
@austincrim Hey, thanks! But I afraid not for the printed, please look [here](https://github.com/MichalLytek/typegraphql-prisma/blob/main/src/generator/helpers.ts#L251) and double check. Never mind if I mistaken it 👍

**Reply by janpio** [reply]
It looks to the me that this is doing the opposite thing of what it should be doing.

This is what the comment looks like in the raw data in Dmmf:
![image](https://user-images.githubusercontent.com/183673/147063729-9d56a691-2606-448b-93a1-c05ca18a9491.png)

- `\\n` is an escaped line break, that should not be touched when printing but replaced with a string `\n` that is output (to get back to what we input via the comment in the schema file).
- `\n` (with no additional `\` before) would be a real line break one, that should be rendered as such.

Correct?

**Reply by j-deng** [reply]
@janpio Hi! Thank you for the reply! But I still confuse, I get the document from `prisma._dmmf` and the code is `(prisma._dmmf.datamodel as DMMF.Datamodel).models`, maybe we have different Object?

![image](https://user-images.githubusercontent.com/4717535/147076344-2844bc91-1906-4d1a-8eb3-1bdd5951fbe4.png)

By the way, I just `console.log` for it, will be shock if I didn't understand what your talk above 😩

**Reply by janpio** [reply]
I think the screenshot I shared is from a test fixture outside of JavaScript context - just the JSON string.
The representation you shared has an additional layer of escaping (my `\\n` is `\\\n`, and my`\n` is `\\n`).
But it still means one of the line breaks has one more `\` which means it is escaped and should not actually be a line break but the literal string `\n` inside the comment - as it was in the original input:
```
/// test\ncomment
/// test too
```

**Reply by j-deng** [reply, marked as answer]
@janpio OK, I got you now, Thanks! From the beginning I was tired to parse the document as file content, which splited by LF. It's different here, no problem. 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
